### PR TITLE
Remove use of obsolete JSON input files for git bakery tasks

### PR DESCRIPTION
**Pull request by rnathuji**
This PR includes the following changes to bakery tasks for the git pipeline:

* `fetch-group`
    * Injects `revised` and `version` data into book collection file.
    * No longer downloads `book-slugs.json`
* `assemble-group`
    * Remove use of book `metadata.json`
* `bake-meta-group`
    * Remove use of book `metadata.json` and `book-slugs.json` (this includes no longer injecting metadata in the task bash script)
    * Updating python script to be compatible with both archive and git flows
* `link-single`
    * Remove use of `book-slugs.json` and instead use data in the group baked files
    * Some minor refactoring of the Python script
* `jsonify-single`
    * Add / use a dedicated book schema

### CODE REVIEW

File: bakery/src/scripts/link_single.py [hunk 1]
```diff
@@ -35,6 +35,21 @@ def create_canonical_map(input_dir):
     return canonical_map
 
 
+def create_slug_map(input_dir):
+    """Create a slug:uuid map from baked metadata files"""
+    baked_metadata_files = Path(input_dir).glob("*.baked-metadata.json")
```
Comment (tomjw64) [inline bakery/src/scripts/link_single.py]: Seems like we are walking the file system twice (see line with `for root, _, files in os.walk(baked_meta_dir):`) to get the same metadata files.

Comment (tomjw64) [reply bakery/src/scripts/link_single.py]: Aha, you fixed it in the next commit. Disregard.

File: bakery/src/scripts/fetch_update_metadata.py [hunk 2]
```diff
@@ -23,33 +26,77 @@ def add_metadata_entries(cnxml_doc, new_metadata):
         metadata.append(element)
 
 
+def determine_book_version(reference, repo, commit):
```
[the remark this answers is not in the document]
Comment (rnathuji) [reply bakery/src/scripts/fetch_update_metadata.py:29]: Oh, I don't believe so -- Maybe you're remembering this from when you [looked at / provided feedback](https://github.com/openstax/cnx/issues/1302#issuecomment-749227820) on it before break?

[the remark this answers is not in the document]
Comment (tomjw64) [reply bakery/src/scripts/fetch_update_metadata.py:29]: Ah, thank you. I was driving myself crazy trying to find that discussion.

File: bakery/src/scripts/fetch_book_group.sh [hunk 3]
```diff
@@ -19,7 +19,6 @@ fi
 fetch-update-meta "${CONTENT_OUTPUT}/raw/.git" "${CONTENT_OUTPUT}/raw/modules" "${CONTENT_OUTPUT}/raw/collections" "$reference"
 rm -rf "${CONTENT_OUTPUT}/raw/.git"
 rm -rf "$creds_dir"
-wget "${BOOK_SLUGS_URL}" -O "${CONTENT_OUTPUT}/book-slugs.json"
```
[the remark this answers is not in the document]
Comment (rnathuji) [reply bakery/src/scripts/fetch_book_group.sh:22]: Good catch! -- [Done](https://github.com/openstax/output-producer-service/pull/270/commits/7160da2c4062dbb3b99ead727c9b5cd2ea2e4737)

**Review by tomjw64 — APPROVED** [review]

**Pull request merged**